libtrip-notification/src/main/java/com/mapbox/navigation/trip/notification/MapboxTripNotification.kt

```diff
@@ -9,33 +9,55 @@ import android.content.Context
 import android.content.Intent
 import android.content.IntentFilter
 import android.os.Build
+import android.text.SpannableString
+import android.text.TextUtils
+import android.text.format.DateFormat
 import android.widget.RemoteViews
 import androidx.core.app.NotificationCompat
 import androidx.core.content.ContextCompat
 import com.mapbox.annotation.navigation.module.MapboxNavigationModule
 import com.mapbox.annotation.navigation.module.MapboxNavigationModuleType
+import com.mapbox.navigation.base.banner.BannerInstruction
+import com.mapbox.navigation.base.options.NavigationOptions
+import com.mapbox.navigation.base.route.model.LegStepNavigation
+import com.mapbox.navigation.base.route.model.RouteProgressNavigation
 import com.mapbox.navigation.base.trip.TripNotification
-import com.mapbox.navigation.base.trip.model.RouteProgress
+import com.mapbox.navigation.trip.notification.utils.distance.DistanceFormatter
+import com.mapbox.navigation.trip.notification.utils.time.TimeFormatter.formatTime
 import com.mapbox.navigation.utils.END_NAVIGATION_ACTION
 import com.mapbox.navigation.utils.NAVIGATION_NOTIFICATION_CHANNEL
 import com.mapbox.navigation.utils.NOTIFICATION_CHANNEL
 import com.mapbox.navigation.utils.NOTIFICATION_ID
 import com.mapbox.navigation.utils.SET_BACKGROUND_COLOR
+import com.mapbox.navigation.utils.extensions.ifNonNull
+import java.util.Calendar
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.channels.ClosedReceiveChannelException
+import kotlinx.coroutines.channels.ClosedSendChannelException
+import kotlinx.coroutines.channels.ReceiveChannel
 
 @MapboxNavigationModule(MapboxNavigationModuleType.TripNotification, skipConfiguration = true)
 class MapboxTripNotification(
```

Comment:
```suggestion
class MapboxTripNotification internal constructor(
    private val applicationContext: Context,
    private val distanceFormatter: DistanceFormatter,
    navigationOptionsBuilder: NavigationOptions
) : TripNotification {
    
    constructor(
        applicationContext: Context,
        navigationOptions: NavigationOptions
    ) : this(applicationContext, DistanceFormatter(...), navigationOptions)
```

Where `NavigationNotificationProvider` is also an `object`.

Reply:
👍